packages/components/src/border-control/types.ts

```diff
@@ -0,0 +1,160 @@
+export type Border = {
+	color?: string;
+	style?: string;
+	width?: string;
+};
+
+export type Color = {
+	name: string;
+	color: string;
+};
+
+export type BorderControlProps = {
+	/**
+	 * An array of color definitions. This may also be a multi-dimensional array
+	 * where colors are organized by multiple origins.
+	 */
+	colors?: Color[];
+	/**
+	 * This toggles the ability to choose custom colors.
+	 */
+	disableCustomColors?: boolean;
+	/**
+	 * This controls whether the alpha channel will be offered when selecting
+	 * custom colors.
+	 */
+	enableAlpha?: boolean;
+	/**
+	 * Provides control over whether the label will only be visible to screen
+	 * readers.
+	 */
+	hideLabelFromVision?: boolean;
+	/**
+	 * This flags the `BorderControl` to render with a more compact appearance.
+	 * It restricts the width of the control and prevents it from expanding to
+	 * take up additional space.
+	 */
+	isSmall?: boolean;
+	/**
+	 * If provided, a label will be generated using this as the content.
+	 */
+	label?: string;
+	/**
+	 * A callback function invoked when the border value is changed via an
+	 * interaction that selects or clears, border color, style, or width.
+	 */
+	onChange: ( value: Border | undefined ) => void;
+	/**
+	 * If opted into, sanitizing the border means that if no width or color have
+	 * been selected, the border style is also cleared and `undefined`
+	 * is returned as the new border value.
+	 */
+	shouldSanitizeBorder?: boolean;
+	/**
+	 * This controls whether to include border style options within the
+	 * `BorderDropdown` sub-component.
+	 */
+	showStyle?: boolean;
+	/**
+	 * An object representing a border or `undefined`. Used to set the current
+	 * border configuration for this component.
+	 */
+	value?: Border;
+	/**
+	 * Controls the visual width of the `BorderControl`.
+	 */
+	width?: string;
+	/**
+	 * Flags whether this `BorderControl` should also render a `RangeControl`
+	 * for additional control over a border's width.
+	 */
+	withSlider?: boolean;
+	/**
+	 * This is passed on to the color related sub-components which need to be
+	 * made aware of whether the colors prop contains multiple origins.
+	 */
+	__experimentalHasMultipleOrigins?: boolean;
+	/**
+	 * This is passed on to the color related sub-components so they may render
+	 * more effectively when used within a sidebar.
+	 */
+	__experimentalIsRenderedInSidebar?: boolean;
```

Comment:
Using experimental props is fine when we need to experiment on something or fix an urgent bug, but we need to have a plan about what to do with them. 

Discussing this with @mirka , we were thinking about how to avoid a proliferation of "forever experimental" props, and we agreed that such props should either be promoted to stable, or be removed after a while.

What do you think about this strategy in general, and what do you think about the evolution of these props? What problem do they solve? Is it a temporary fix while we come up with a better solution somewhere else (e.g. in the block editor package)?

Reply:
These props were added to pass through to the `ColorPalette` component itself. Whatever its strategy for removal or stabilisation of those same props would dictate them here too.